File: BirdWatcher/src/main/java/com/birdwatcher/controller/SignupController.java
package com.birdwatcher.controller;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

import com.birdwatcher.model.User;
import com.birdwatcher.model.UserDTO;
import com.birdwatcher.model.UserDirectory;

@Controller
@RequestMapping(value="/signup")
public class SignupController {
	@RequestMapping(method = RequestMethod.POST)
	@ResponseBody
    public String registerUser(@RequestBody User activeUser){
		System.out.println("Inside signup Controller");
		String username=activeUser.getUsername();
		String email=activeUser.getEmail();
		String password=activeUser.getPassword();
		UserDirectory users = new UserDirectory();
		User newUser= new User();
		users.addNewUser(username, password, email);
		System.out.print(users.getUserDirectory());
		return "success";
    }
}

File: BirdWatcher/src/main/resources/static/js/signupController.js
app.controller('signupController', function($scope,$location,$http){	
	$scope.signup=function(){
		var userData={
				username:$scope.username,
				email:$scope.email,
				password:$scope.password
		};
		$http({
			method:"POST",
			url:"/signup",
			data:userData
		}).then(function success(response){
			console.log("Signup data sent successfully");
			if(response.data=="success"){
				console.log("User Added");
				alert("User added successfully");
				$location.path('/home');
			}else{
				console.log("User not added");
				$location.path('/signup');
			}
		}, function error(response){
			console.log("Signup data not sent");
			$location.path('/signup');
		});
	}
});
